packages/localize/src_client/lit-localize.ts

```diff
@@ -346,4 +348,24 @@ export function _msg(
   return template;
 }
 
+const hashCache = new Map<TemplateStringsArray | string, string>();
+
+function generateId(template: TemplateLike): string {
+  if (typeof template === 'function') {
+    const numParams = template.length;
+    // Note that by using HASH_DELIMITER as the template parameter here, we can
+    // skip splitting and re-joining when we perform the hash.
+    const params = Array(numParams).fill(HASH_DELIMITER);
+    template = template(...params);
```

Comment:
Another option here, that would allow expressions in templates, would be to evaluate the template function with the provided args, then pull the strings from the template result.

Reply:
> fascinating :) - it's safe to pass this here because expressions are required to be simple identifiers that only reference the parameters. Might be nice to comment that for future maintainers

Done.

> Another option here, that would allow expressions in templates, would be to evaluate the template function with the provided args, then pull the strings from the template result.

Yeah, that occurred to me while working on this too! I like idea, but there are a few details to account for I thought of:

1. Calls like ``` msg(`Hello ${who}!`) ``` won't work, because we'd just receive a string. I think we'd need to vend a tag like ``` str`Hello ${who}!` ``` for this case so that we can capture the strings and values.

2. The order of placeholders can change during translation, so we'd need to analyze their new locations and include that re-ordering logic into the functions we generate.

3. A side-effect advantage of identifiers is that I think they'll tend to be easier for translators to understand, since they'll be more self-describing. Translators see the expressions (but can't change them), to help them understand context. I think churn in the values is also less likely with identifiers.